scripts/tagging/parseversion.py

```diff
@@ -52,18 +52,18 @@ def _parseVersion( self ):
       raise self.VersionStringError("Cannot parse this version string: %s" % self.version)
 
     if len(parts) >= 3:
-      if 'pre' == parts[2]:
+      if 'pre' == parts[2] or 'p' == parts[2]:
         self.pre = 0
-      elif 'pre' in parts[2]:
-        self.pre = int(parts[2].strip('pre'))
+      elif 'pre' in parts[2] or 'p' in parts[2]:
+        self.pre = int(parts[2].strip('pre').strip('p'))
       else:
         self.patch = int( parts[2] )
```

Comment:
```suggestion
        self.patch = int( parts[2].strip('p') )
```
Would this by itself solve the ILDConfig problem?

Reply:
Yes, indeed that would work as a minimal solution as well. But that basically ignores `p` releases(?). At least the following breaks (but it also does with all other changes in place)

```bash
python ilcsofttagger.py --lastTag v02-02-p01 --packages ILDConfig
```

Reply:
In what way does it break? Is it because there is also 02-02-01 (https://github.com/iLCSoft/ILDConfig/releases/tag/v02-02-01)

Reply:
This is not strictly what initially triggered this pull request, and a bit contrived for the current situation, but if `p` releases should be considered patch releases they should be usable as arguments to `--lastTag` I think.

So with the following
```bash
python ilcsofttagger.py --lastTag v02-02-p01 --ignoreMissingCmake --properRelease --packages ILDConfig
```

I get a parsing error without any changes (this is what I intended to fix with this pull request):
```console
INFO  - Tagger  : Checking package: ILDConfig
ERROR - Version : Failed to parse version: invalid literal for int() with base 10: 'p02'
ERROR - Tagger  : Failure for iLCSoft/ILDConfig/master/None
ERROR - Tagger  : ERROR: broken version string
ERROR - root    : Error during runtime: ERROR
```

Including the minimal fix with stripping the `p` I get:
```console
INFO  - Tagger  : Checking package: ILDConfig
INFO  - ILDConfig: Found latest tag v02-02-p01
INFO  - ILDConfig: Set new version: v02-02-02
ERROR - ILDConfig: Required (pre-)version ('v02-02-02') is the same as or older than the last tag('v02-02-p01')
ERROR - Tagger  : Failure for iLCSoft/ILDConfig/master/None
ERROR - Tagger  : ERROR: Invalid version required
ERROR - root    : Error during runtime: ERROR
```

Looking at the debug output, it looks like the `-pXX` should always populate the last field of the version tuple, currently it populates the one at which it appears, which seems to break version comparison (?).

Reply:
What should be the outcome of the command?

Looking at the code the issue is with
https://github.com/iLCSoft/iLCInstall/blob/7e87bd1d301dbd13517a870f302c5ea7b2a83ae5/scripts/tagging/helperfunctions.py#L37

called from
https://github.com/iLCSoft/iLCInstall/blob/7e87bd1d301dbd13517a870f302c5ea7b2a83ae5/scripts/tagging/gitinterface.py#L94-L99
and not the parsing of the version string.

Reply:
In this case the tags are the following (already sorted):
```
v02-02
v02-02-p01
v02-02-p02
v02-02-01
```
So when I specify `--lastTag v02-02-p01` ideally all changes from `v02-02-p02` and `v02-02-01` should be collected.

Reply:
What could work in principle is to parse the `pXY` into the `Version.pre` field for comparison. The `Version.isPre` could in principle still be used for differentiating between pre releases and patch releases. On the other hand I am not entirely sure if is worth all the complexity for handling this arguable edge case.

Reply:
The real fun begins when you get versions like https://github.com/iLCSoft/ILDConfig/releases/tag/v01-19-06-p01
I wouldn't abuse pre for this, because in other places pre is used only for the pre purpose.
One could add a second patch member variable to store entries with `p`

Reply:
Yeah, after playing with this a bit, abusing the `pre` field doesn't really work out, even if the parsing and sorting problems are sorted out, there are still some remaining problems with the `str` conversion.

So I think fixing this properly would take a bit of time. However, simply fixing the parsing, would at least allow to use the tagging script to tag `ILDConfig`, even if `p` releases are not handled completely correctly.